### PR TITLE
Vesting: add position termination functionality

### PR DESCRIPTION
**Pull request by larry0x**
Add an owner-only function `terminate_position`. All tokens that have already been vested will remain in the position. All unvested tokens will be sent to the owner.

### CODE REVIEW

File: contracts/vesting/src/contract.rs [hunk 1]
```diff
@@ -113,6 +116,50 @@ pub fn create_position(
         .add_attribute("duration", vest_schedule.duration.to_string()))
 }
 
+pub fn terminate_position(
```
Comment (piobab) [inline contracts/vesting/src/contract.rs:119]: Can be private function

File: contracts/vesting/src/contract.rs [hunk 2]
```diff
@@ -113,6 +116,50 @@ pub fn create_position(
...
+    env: Env,
+    info: MessageInfo,
+    user_addr: Addr,
+) -> StdResult<Response> {
+    let current_time = env.block.time.seconds();
+
+    // only owner can terminate allocations
+    let owner_addr = OWNER.load(deps.storage)?;
+    if info.sender != owner_addr {
+        return Err(StdError::generic_err("only owner can terminate allocations"));
```
Comment (piobab) [inline contracts/vesting/src/contract.rs:130]: Maybe return standard one - Unauthorized error

**Review by piobab — APPROVED** [review]

File: contracts/vesting/src/contract.rs (hunk 2 above)
Comment (larry0x) [reply contracts/vesting/src/contract.rs:130]: `StdError::Unauthorized` doesn't give any explanation on what action is unauthorized and why it's unauthorized, which i don't find very useful. ideally we should define a `ContractError` here, but that's out of the scope of this PR.

File: contracts/vesting/src/contract.rs (hunk 1 above)
Comment (larry0x) [reply contracts/vesting/src/contract.rs:119]: this only means whether someone who imports this crate into their project can use the function. it doesn't have anything to do with how the contract works. for the contract, only the `entry_point` functions are exposed.

Comment (piobab) [reply contracts/vesting/src/contract.rs:119]: Yes and I mean that you don't expose your internal logic. If you prefer to leave pub - no problem.

**Review by dancreee — APPROVED** [review]

**Comment by larry0x** [comment]
> @larry0x can we merge so I will send to reaudit?

merged!

**Pull request merged**